### PR TITLE
RAP-4097 - use 1:1 mapping between search type and scheme (plus meta and uri), and auto-generate it

### PR DESCRIPTION
**Pull request by dukenguyen**

### CODE REVIEW

File: Libs/RaptureAddins/ElasticSearch/src/test/java/rapture/elasticsearch/ElasticSearchSearchRepositoryTest.java [hunk 1]
```diff
@@ -150,117 +142,133 @@ public void testSearchAndConvert() {
...
 
-        r = e.search(Arrays.asList(SearchRepoType.SERIES.toString()), "4.0");
+        r = e.search(Arrays.asList(Scheme.SERIES.toString()), "4.0");
         assertEquals(1, r.getTotal().longValue());
 
         e.remove(new RaptureURI("series://removerepo/a/b/c"));
         e.refresh();
 
-        r = e.search(Arrays.asList(SearchRepoType.SERIES.toString()), "v3");
+        r = e.search(Arrays.asList(Scheme.SERIES.toString()), "v3");
```
Comment (scarabus) [inline Libs/RaptureAddins/ElasticSearch/src/test/java/rapture/elasticsearch/ElasticSearchSearchRepositoryTest.java]: You should search for "4.0" not "v3" here to prove that it works

Comment (dukenguyen) [inline Libs/RaptureAddins/ElasticSearch/src/test/java/rapture/elasticsearch/ElasticSearchSearchRepositoryTest.java]: i'm already doing that 2-3 lines above.  this is a test for a string that should not be found, as can be seen by the assertion of 0 right after

Comment (scarabus) [inline Libs/RaptureAddins/ElasticSearch/src/test/java/rapture/elasticsearch/ElasticSearchSearchRepositoryTest.java]: I understand that. You confirm that 4.0 is there. Then you delete the series, but you don't test that 4.0 has gone; you only test for the absence of something that was never there.

Comment (dukenguyen) [inline Libs/RaptureAddins/ElasticSearch/src/test/java/rapture/elasticsearch/ElasticSearchSearchRepositoryTest.java]: yea you're right, thanks for adding in your pr.

**Pull request merged**